src/libraries/System.Net.Http/src/System/Net/Http/Headers/KnownHeader.cs

```diff
@@ -47,5 +47,7 @@ public KnownHeader(string name, HttpHeaderType headerType, HttpHeaderParser? par
         public string[]? KnownValues { get; }
         public byte[] AsciiBytesWithColonSpace { get; }
         public HeaderDescriptor Descriptor => new HeaderDescriptor(this);
+
+        public string Separator => Parser is HttpHeaderParser parser && parser.SupportsMultipleValues ? parser.Separator! : HttpHeaderParser.DefaultSeparator;
```

Comment:
This is beyond the scope of this PR:

I don't understand why Parser.Separator doesn't just do this, so we don't have to check for SupportsMultipleValues here explicitly. There are probably similar places we have this logic and it seems better to just encapsulate it in parser.Separator so that the right thing just happens transparently.

Reply:
Looks like this was now the only place that called `HttpHeaderParser.Separator`. Simplified the logic in `HttpHeaderParser` so that we can avoid this check here.